isis/tests/FunctionalTestsChan1m32isis.cpp

```diff
@@ -0,0 +1,550 @@
+  ASSERT_EQ(double(pixels["Multiplier"]), 1.0);
+
+  // Instrument Group
+  PvlGroup &inst = isisLabel->findGroup("Instrument", Pvl::Traverse);
+  ASSERT_EQ(inst["SpacecraftName"][0].toStdString(), "CHANDRAYAAN-1");
+  ASSERT_EQ(inst["InstrumentId"][0].toStdString(), "M3" );
+  ASSERT_EQ(inst["TargetName"][0].toStdString(), "MOON" );
+  ASSERT_EQ(inst["SpacecraftClockStartCount"][0].toStdString(), "4/1165041.799" );
+  ASSERT_EQ(inst["SpacecraftClockStopCount"][0].toStdString(), "4/1165065" );
+  ASSERT_DOUBLE_EQ(double(inst["LineExposureDuration"]), 101.76);
+  ASSERT_DOUBLE_EQ(double(inst["SpatialSumming"]), 2);
+  ASSERT_EQ(inst["StartTime"][0].toStdString(), "2009-01-06T11:34:23" );
+  ASSERT_EQ(inst["StopTime"][0].toStdString(), "2009-01-06T11:34:47" );
+  ASSERT_EQ(inst["SpacecraftYawDirection"][0].toStdString(), "UNKNOWN" );
+  ASSERT_EQ(inst["OrbitLimbDirection"][0].toStdString(), "UNKNOWN" );
+
+  // Archive Group
+  PvlGroup &archive = isisLabel->findGroup("Archive", Pvl::Traverse);
+  ASSERT_EQ(archive["ProductId"][0].toStdString(), "M3G20090106T113423_V01_L0" );
+  ASSERT_EQ(archive["ProductType"][0].toStdString(), "RAW_IMAGE" );
+
+  // BandBin Group
+  // Check size, first, 2 middle, and last values? Enough?
+  PvlGroup &bandbin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  ASSERT_EQ(bandbin["Center"].size(), 85);
+  ASSERT_EQ(bandbin["FilterNumber"].size(), 85);
+  ASSERT_EQ(bandbin["OriginalBand"].size(), 85);
+
+  ASSERT_DOUBLE_EQ(bandbin["Center"][0].toDouble(), 460.990);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][21].toDouble(), 1009.95);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][42].toDouble(), 1429.15);
+  ASSERT_DOUBLE_EQ(bandbin["Center"][84].toDouble(), 2976.20);
+
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][0].toDouble(), 5);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][21].toDouble(), 57);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][42].toDouble(), 99);
+  ASSERT_DOUBLE_EQ(bandbin["FilterNumber"][84].toDouble(), 253);
+
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][0].toDouble(), 1);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][21].toDouble(), 22);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][42].toDouble(), 43);
+  ASSERT_DOUBLE_EQ(bandbin["OriginalBand"][84].toDouble(), 85);
+
+  // Kernels Group
+  PvlGroup &kernel = isisLabel->findGroup("Kernels", Pvl::Traverse);
+  ASSERT_EQ(int(kernel["NaifFrameCode"]), -86520);
+
+  Histogram *hist = cube.histogram();
+
+  ASSERT_NEAR(hist->Average(), 776.031, .001);
+  ASSERT_EQ(hist->Sum(), 1241649);
+  ASSERT_EQ(hist->ValidPixels(), 1600);
+  ASSERT_NEAR(hist->StandardDeviation(), 449.337, .001);
+
+  delete hist;
+}
+
+
+TEST(Chan1m32Isis, Chan1m32IsisTestBadFile) {
+  QTemporaryDir prefix;
+  QString cubeFileName = prefix.path() + "/chan1m32isisTEMP.cub";
+  QVector<QString> args = {"from=data/chan1m32isis/badFile/bad.file",
+                           "to=" + cubeFileName };
+
+  UserInterface options(APP_XML, args);
+  ASSERT_THROW(chan1m32isis(options), IException);
```

Comment:
@Kelvinrr It seems like there are multiple ways we are checking errors. I think we should come up with a standard.

Reply:
I think sticking with try{do_something; FAIL()}Catch(){e.what substring_check} is the way to go, it allows you to assert both the error type and the message. Especially for cases like this where the exception type is not enough to reasonably assume that the Exception was thrown for the right reason. Here, IException can be triggered by a number of things that may not be what you want.